File: fileManager.js
const Discord = require('discord.js');
const client = new Discord.Client();
const fs = require('fs')

const getAllDirFiles = function(dirPath, arrayOfFiles) {
    files = fs.readdirSync(dirPath)
  
    arrayOfFiles = arrayOfFiles || []
  
    files.forEach(function(file) {
      if (fs.statSync(dirPath + "/" + file).isDirectory()) {
        arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles)
      } else {
        arrayOfFiles.push(file)
      }
    })
  
    return arrayOfFiles
}

const sendImage = function (num, path, fileName, message, lastNum, color) {
  var title = "" + fileName + num;
    lastNum = num;
    if (num <= 9) {
      fileName = fileName + "0" + num;
    } else {
      fileName = fileName + num;
    }
    if(fs.existsSync(path + fileName + ".gif")){
      fileName = fileName + ".gif"
      console.log(fileName)
      var embed = new Discord.MessageEmbed()
        .setTitle(title.toUpperCase())
        .setColor(color)
        .attachFiles([path + fileName])
        .setImage('attachment://' + fileName);
        return embed
    } else if (fs.existsSync(path + fileName + ".png")) {
      fileName = fileName + ".png"
      console.log(fileName)
      var embed = new Discord.MessageEmbed()
        .setTitle(title.toUpperCase())
        .setColor(color)
        .attachFiles([path + fileName])
        .setImage('attachment://' + fileName);
        return embed
    } else if (fs.existsSync(path + fileName + ".jpg")) {
        fileName = fileName + ".jpg"
        console.log(fileName)
        var embed = new Discord.MessageEmbed()
        .setTitle(title.toUpperCase())
        .setColor(color)
        .attachFiles([path + fileName])
        .setImage('attachment://' + fileName);
        return embed
    } else if (fs.existsSync(path + fileName + ".mp4")) {
        fileName = fileName + ".mp4"
        console.log(fileName)
        var embed = {files: [(path + fileName)]}
        return embed
    } else if (fs.existsSync(path + fileName + ".mov")) {
      fileName = fileName + ".mov"
      console.log(fileName)
      var embed = {files: [(path + fileName)]}
        return embed
    } else {
      console.log("Error " + path + fileName + " not found...")
      message.channel.send("This ain't it chief. I couldn't find the gif.");
    }
  };

const getItem = function(folder) {
  console.log("Checking " + folder + " for items...")
  var num = Math.floor(Math.random() * getAllDirFiles(folder).length)
  var counter = 0
  var item
  fs.readdirSync(folder).forEach(file => {
    //console.log(file)
    //console.log("Counter (" + num + "): " + counter)
    if (counter == num) {
      item = file
      //console.log(item + " has been saved...")
    }
    counter++
    console.log("Item to return: " + item)
    return item;
  });
}

module.exports.getAllDirFiles = getAllDirFiles;  
module.exports.sendImage = sendImage;
module.exports.getItem = getItem;

File: fileManager.py
import discord
from re import search
import random
from loggingChannel import sendLog
import os.path
from os import path
from discord.ext import commands


async def sendImage(message, client, fileName, num, dir):
    found = True
    if num <= 9:
        fileName = fileName + "0" + str(num)
    else:
        fileName = fileName + str(num)

    if path.isfile(dir + fileName + ".gif"):
        print(f'Found {(dir + fileName + ".gif")}')
        file = discord.File(dir + fileName + ".gif", filename="image.gif")
    elif path.isfile(dir + fileName + ".png"):
        print(f'Found {(dir + fileName + ".png")}')
        file = discord.File(dir + fileName + ".png", filename="image.png")
    elif path.isfile(dir + fileName + ".jpg"):
        print(f'Found {(dir + fileName + ".jpg")}')
        file = discord.File(dir + fileName + ".jpg", filename="image.jpg")
    elif path.isfile(dir + fileName + ".mp4"):
        print(f'Found {(dir + fileName + ".mp4")}')
        file = discord.File(dir + fileName + ".mp4", filename="video.mp4")
    elif path.isfile(dir + fileName + ".mov"):
        print(f'Found {(dir + fileName + ".mov")}')
        file = discord.File(dir + fileName + ".mov", filename="video.mov")
    else:
        print(await sendLog(log=f'This ain\'t it chief. I couldn\'t find the gif.', client=client))
        found = False

    if found:
        await message.channel.send(file=file)


async def sendImageNew(interaction, client, fileName, num, dir):
    found = True
    if num <= 9:
        fileName = fileName + "0" + str(num)
    else:
        fileName = fileName + str(num)

    if path.isfile(dir + fileName + ".gif"):
        print(f'Found {(dir + fileName + ".gif")}')
        file = discord.File(dir + fileName + ".gif", filename="image.gif")
    elif path.isfile(dir + fileName + ".png"):
        print(f'Found {(dir + fileName + ".png")}')
        file = discord.File(dir + fileName + ".png", filename="image.png")
    elif path.isfile(dir + fileName + ".jpg"):
        print(f'Found {(dir + fileName + ".jpg")}')
        file = discord.File(dir + fileName + ".jpg", filename="image.jpg")
    elif path.isfile(dir + fileName + ".mp4"):
        print(f'Found {(dir + fileName + ".mp4")}')
        file = discord.File(dir + fileName + ".mp4", filename="video.mp4")
    elif path.isfile(dir + fileName + ".mov"):
        print(f'Found {(dir + fileName + ".mov")}')
        file = discord.File(dir + fileName + ".mov", filename="video.mov")
    else:
        print(await sendLog(log=f'This ain\'t it chief. I couldn\'t find the gif.', client=client))
        found = False

    if found:
        await interaction.response.send_message(file=file)
